assets/js/src/newsletters/listings/mixins.jsx

```diff
@@ -401,134 +401,6 @@ const MailerMixin = {
-    if (state.meta.mta_method === 'PHPMail') {
-      mailerErrorCheckSettings = (
-        <>
-          <p>{ MailPoet.I18n.t('mailerSendErrorCheckConfiguration') }</p>
-          <br />
-          <p>
-            {
-              ReactStringReplace(
-                MailPoet.I18n.t('mailerSendErrorUseSendingService'),
-                /<b>(.*?)<\/b>/g,
-                (match, key) => <b key={key}>{ match }</b>
-              )
-            }
-          </p>
-          <p>
-            <a
-              href="https://www.mailpoet.com/free-plan/?utm_source=plugin&utm_campaign=sending-error"
-              target="_blank"
-              rel="noopener noreferrer"
-            >
-              { MailPoet.I18n.t('mailerSendErrorSignUpForSendingService') }
-            </a>
-          </p>
-          <br />
-        </>
-      );
-    } else {
-      mailerErrorCheckSettings = (
-        <p>
-          {
-            ReactStringReplace(
-              MailPoet.I18n.t('mailerCheckSettingsNotice'),
-              /\[link\](.*?)\[\/link\]/g,
-              match => <a href="?page=mailpoet-settings#mta" key="check-sending">{ match }</a>
-            )
-          }
-        </p>
-      );
-    }
-
-    // append error code if available
-    let mailerErrorMessage = state.meta.mta_log.error.error_message;
-    if (state.meta.mta_log.error.error_code) {
-      mailerErrorMessage += mailerErrorMessage ? ', ' : '';
-      mailerErrorMessage += MailPoet.I18n.t('mailerErrorCode').replace('%$1s', state.meta.mta_log.error.error_code);
-    }
-
-    return (
-      <div>
-        <p>
-          {
-            state.meta.mta_log.error.operation === 'send'
-              ? MailPoet.I18n.t('mailerSendErrorNotice').replace('%$1s', state.meta.mta_method)
-              : MailPoet.I18n.t('mailerConnectionErrorNotice')
-          }
-          :
-          {' '}
-          <i>{ mailerErrorMessage }</i>
-        </p>
-        { mailerErrorCheckSettings }
-        <p>
-          <a
-            href="javascript:;"
-            className="button button-primary"
-            onClick={MailerMixin.resumeMailerSending}
-          >
-            { MailPoet.I18n.t('mailerResumeSendingButton') }
-          </a>
-        </p>
-      </div>
-    );
-  },
-  getMailerErrorType(state) {
-    if (state.meta.mta_log.error.operation === 'migration') {
-      return 'system';
-    }
-    return 'error';
-  },
-  resumeMailerSending() {
-    MailPoet.Ajax.post({
-      api_version: window.mailpoet_api_version,
-      endpoint: 'mailer',
-      action: 'resumeSending',
-    }).done(() => {
-      MailPoet.Notice.hide('mailpoet_mailer_error');
-      MailPoet.Notice.success(MailPoet.I18n.t('mailerSendingResumedNotice'));
-      window.mailpoet_listing.forceUpdate();
-    }).fail((response) => {
-      if (response.errors.length > 0) {
-        MailPoet.Notice.error(
-          response.errors.map(error => error.message),
-          { scroll: true }
-        );
-      }
-    });
```

Comment:
So if I understand this correctly:
1. Originally this used to show error notice if error found on `checkMailerStatus` call and then render its body. This is called from multiple places.
2. Now it only shows the notice and expects the body will be rendered by `MailerError` using `{ this.state.meta.mta_method && <MailerError {...this.state.meta} /> }` line.

Is that right? Maybe I just misunderstood.

Now, doesn't this change behavior significantly? To have the body actually rendered it's necessary that `listings.jsx` is rendered? How is that guaranteed?

Also, it would be nice to get rid of mixins but doesn't this make the code actually worse? Rendering notice without body and hoping it will appear there feels really not good.

Reply:
You're misunderstanding, let me explain. The `MailerMixin.checkMailerStatus` was doing the following:

1- if there is an authorization error, it shows a notice with id `mailpoet_authorization_error`.
2- if the sending is paused due to an other type of sending errors, it creates a container with id `mailpoet_mailer_error` if not present, then renders some JSX into that container.

I kept the first notice (1) in the Mixin because it does not relate to the task and moved the second one. Now the mixin will still be called to do (1), while the part (2) will be done in pure React (JSX is rendered directly into the component, no container is created).

Does that make sense?

Reply:
Right, I was wrong with the body, `mailpoet_authorization_error` is another notice, not a wrapper.

However, the change of behavior is still there - rendering of  `mailpoet_mailer_error` was originally triggered by `checkMailerStatus`, not anymore. This means that it can (maybe) happen that the `mailpoet_mailer_error` will not render when it should - `checkMailerStatus` was originally called from various places. Are *all* of them followed by render of `MailerError` in the current implementation? Just double checking.

Reply:
`checkMailerStatus` is called in the 4 emails listings. they all render the `Listing` component and pass a callback which will call `checkMailerStatus` after loading emails data. With the current implementation, the `Listing` component will render `MailerError` itself whenever it's rendered and there is some sending error in the meta data loaded from the API.
You may say: "But we are using `Listing` on other pages like Forms and Subscribers, we don't want the `MailerError` to render there!". That's correct, but notice that the API only includes `mta_log` and `mta_method` on the meta if we requested newsletters. So `MailerError` will always render `null` on other pages.